### PR TITLE
Update conf test readme

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Update conf test readme
Added the supported operations comments in each test.yml

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #411 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/conformance/README.md [hunk 1]
```diff
@@ -1,23 +1,60 @@
...
+	switch tc.Component {
+	case "azure.servicebusqueues":
+		binding = b_azure_servicebusqueues.NewAzureServiceBusQueues(testLogger)
+	case "azure.storagequeues":
+		binding = b_azure_storagequeues.NewAzureStorageQueues(testLogger)
+	case "azure.eventgrid":
+		binding = b_azure_eventgrid.NewAzureEventGrid(testLogger)
+	case "kafka":
+		binding = b_kafka.NewKafka(testLogger)
+        case "new-component":
```
Comment (mukundansundar) [inline tests/conformance/README.md:35]: indentation looks correct in rich diff.

File: tests/config/bindings/tests.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## Supported operations: create, operations and read
```
Comment (tcnghia) [inline tests/config/bindings/tests.yml:1]: Why doubling the `#`?

File: tests/config/bindings/tests.yml [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+## All config keys prefixed with output_ will be part of request for the output binding
+## readBindingTimeout : timeout to wait to receive test event
+## url: specific to http component, url of the http server
+## method: what method to use
```
Comment (tcnghia) [inline tests/config/bindings/tests.yml:5]: Indent the parameters?

File: tests/conformance/README.md (hunk 1 above)
Comment (tcnghia) [reply tests/conformance/README.md:35]: Maybe a tab vs spaces issue

File: tests/conformance/README.md [hunk 4]
```diff
@@ -1,23 +1,60 @@
...
-8. Each `component type` has its own `_test` file to trigger the conformance tests.
-9. Each test added will also need to be added to the `conformance.yml` workflow file. 
+8. Each `component type` has its own `_test` file to trigger the conformance tests. Eg: `bindings_test.go`.
+9. Each test added will also need to be added to the `conformance.yml` workflow file.
+
+## Conformance test workflow
+1. All components/services needed are currently manually setup. 
+2. If the component is tested locally within a docker container requiring no secrets, then add the component to the `pr-components` step in conformance test workflow `.github/conformance.yml`. 
+3. If the component is tested against a service and requires secrets, then add the component to the `cron-components` step in conformance test workflow `.github/conformance.yml`. 
+4. `pr-components` step generates the component matrix for which the conformance tests will be run on each PR.
```
Comment (tcnghia) [inline tests/conformance/README.md:17]: Merge with 2?

File: tests/conformance/README.md [hunk 5]
```diff
@@ -1,23 +1,60 @@
...
-9. Each test added will also need to be added to the `conformance.yml` workflow file. 
+8. Each `component type` has its own `_test` file to trigger the conformance tests. Eg: `bindings_test.go`.
+9. Each test added will also need to be added to the `conformance.yml` workflow file.
+
+## Conformance test workflow
+1. All components/services needed are currently manually setup. 
+2. If the component is tested locally within a docker container requiring no secrets, then add the component to the `pr-components` step in conformance test workflow `.github/conformance.yml`. 
+3. If the component is tested against a service and requires secrets, then add the component to the `cron-components` step in conformance test workflow `.github/conformance.yml`. 
+4. `pr-components` step generates the component matrix for which the conformance tests will be run on each PR. 
+5. `cron-components` defines the components for which the conformance test will be run against the master code in a scehduled manner.
```
Comment (tcnghia) [inline tests/conformance/README.md:18]: Merge with 3?

File: tests/conformance/README.md [hunk 6]
```diff
@@ -1,23 +1,60 @@
...
+
+## Integrating a new component with conformance tests 
+1. Add the component sepcific YAML to `tests/config/<COMPONENT-TYPE>/<COMPONENT>/<FILE>.yaml`.
```
Comment (tcnghia) [inline tests/conformance/README.md:21]:
```suggestion
1. Add the component specific YAML to `tests/config/<COMPONENT-TYPE>/<COMPONENT>/<FILE>.yaml`.
```

**Review by tcnghia — CHANGES_REQUESTED** [review]

**Review by tcnghia — APPROVED** [review]

File: tests/config/bindings/tests.yml [hunk 7]
```diff
@@ -1,3 +1,9 @@
+# Supported operations: create, operations and read
+# Config map:
+## All config keys prefixed with output_ will be part of request for the output binding
+## readBindingTimeout : timeout to wait to receive test event
+## url: specific to http component, url of the http server
+## method: what method to use
```
Comment (halspang) [inline tests/config/bindings/tests.yml:6]: This is also specific to the http component.

File: tests/config/bindings/tests.yml [hunk 8]
```diff
@@ -1,3 +1,9 @@
+# Supported operations: create, operations and read
```
Comment (halspang) [inline tests/config/bindings/tests.yml:1]: I don't think we've brought in any other components yet, but the bindings as a whole support more than create and the base operations are all included in the test already.

File: tests/conformance/README.md [hunk 9]
```diff
@@ -1,23 +1,58 @@
...
 7. All the tests configurations are defined in `common.go` file.
-8. Each `component type` has its own `_test` file to trigger the conformance tests.
-9. Each test added will also need to be added to the `conformance.yml` workflow file. 
+8. Each `component type` has its own `_test` file to trigger the conformance tests. Eg: `bindings_test.go`.
+9. Each test added will also need to be added to the `conformance.yml` workflow file.
+
+## Conformance test workflow
+1. All components/services needed are currently manually setup. 
+2. If the component is tested locally within a docker container requiring no secrets, then add the component to the `pr-components` step in conformance test workflow `.github/conformance.yml`. `pr-components` step generates the component matrix for which the conformance tests will be run on each PR. 
+3. If the component is tested against a service and requires secrets, then add the component to the `cron-components` step in conformance test workflow `.github/conformance.yml`. `cron-components` defines the components for which the conformance test will be run against the master code in a scehduled manner.
```
Comment (halspang) [inline tests/conformance/README.md:16]: Should this include info on how to setup a component? Points 2 and 3, to me, read as if adding it to cron-components will just make it magically work without any setup. 

If we're the ones who have to setup the component for automated testing, we can just say that too.

**Review by halspang — COMMENTED** [review]
Overall, lgtm. Just a few quick comments.

File: tests/config/bindings/tests.yml (hunk 8 above)
Comment (mukundansundar) [reply tests/config/bindings/tests.yml:1]: yes this is just for now.

File: tests/conformance/README.md (hunk 9 above)
Comment (mukundansundar) [reply tests/conformance/README.md:16]: this is specific to modifying the workflow. 
Also see the first point
`1. All components/services needed are currently manually setup. `

Comment (halspang) [reply tests/conformance/README.md:16]: I still think it'd be good to state that it's the dapr team's responsibility to setup the components for automated tests.

Comment (mukundansundar) [reply tests/conformance/README.md]: Changed ... see now

**Review by halspang — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**